### ISSUE TITLE
Drawing shapes appears to be off...?

### ISSUE THREAD

**Issue by SuperRiderTH** [enigma-dev/enigma-dev#656]
Drawing text and rectangles appear to draw in different locations between Studio and Enigma.
Using the same images from the surface issue.
Studio:
https://dl.dropboxusercontent.com/u/88790752/studiosurf.png
Enigma:
https://dl.dropboxusercontent.com/u/88790752/esurf.png

**Comment by RobertBColton** [comment]
Closing as a duplicate of #244. This is in fact a bug. For one, GMSv1.4 is using Direct3D 9Ex, which has the dreaded [texel centering problems](https://docs.microsoft.com/en-us/windows/desktop/direct3d9/directly-mapping-texels-to-pixels). GMSv1.4 is also using ANGLE which deals with this problem in its vertex shader. Now, I've tried to get ENIGMA D3D9/D3D11 and GL1/GL3 all with the same pixel alignment, and miserably failed even using the ANGLE solution.

Basically, these problems are all related and resulting from API differences in OpenGL and Direct3D. We have to figure out a central unified way of solving the problem (which is easier now that all the batching logic, models, and stuff are generic). For now, this issue is just a duplicate of the other one.
